cfme/services/catalogs/catalog_items/__init__.py

```diff
@@ -298,41 +345,86 @@ def delete_if_exists(self):
         if self.exists:
             self.delete()
 
-    def add_button_group(self):
-        button_name = fauxfactory.gen_alpha()
-        view = navigate_to(self, 'AddButtonGroup')
-        view.fill({'btn_group_text': 'group_text',
-                   'btn_group_hvr_text': button_name,
-                   'btn_image': self.button_icon_name})
+    def add_button_group(self, **kwargs):
+        button_name = kwargs.get("text", "gp_{}".format(fauxfactory.gen_alpha()))
+        hover_name = kwargs.get("hover", "hover_{}".format(fauxfactory.gen_alpha()))
+        image = kwargs.get("image", "broom")
+
+        view = navigate_to(self, "AddButtonGroup")
+        view.fill(
+            {
+                "btn_group_text": button_name,
+                "btn_group_hvr_text": hover_name,
+                "btn_image": image,
+            }
+        )
+
         view.add.click()
-        view = self.create_view(DetailsCatalogItemView)
-        assert view.is_displayed
+        view = self.create_view(DetailsCatalogItemView, wait="10s")
         view.flash.assert_no_error()
         return button_name
 
-    def add_button(self):
-        button_name = fauxfactory.gen_alpha()
+    def button_group_exits(self, name):
+        view = navigate_to(self, "Details")
```

Comment:
@valaparthvi while respecting code should not repeat we are not missing python identity l.e. `readability`? second I don't like some method on which i depend for view. third I don't like to pass its button or group to that method. In case of LOC  ratio will be 2:3 for reducing 4 line I need to pass hard-coded button or group.